pkg/domain/infra/abi/play.go

```diff
@@ -267,112 +265,165 @@ func (ic *ContainerEngine) playKubePod(ctx context.Context, podName string, podY
 		configMaps = append(configMaps, cm)
 	}
 
-	containers := make([]*libpod.Container, 0, len(podYAML.Spec.Containers))
-	cwd, err := os.Getwd()
-	if err != nil {
-		return nil, err
-	}
-	for _, container := range podYAML.Spec.Containers {
-		// Contains all labels obtained from kube
-		labels := make(map[string]string)
-		var pulledImage *libimage.Image
-		buildFile, err := getBuildFile(container.Image, cwd)
+	if podOpt.Infra {
+		imagePull := "k8s.gcr.io/pause:3.5"
+		if podOpt.InfraImage != "k8s.gcr.io/pause:3.5" && podOpt.InfraImage != "" {
+			imagePull = podOpt.InfraImage
+		}
+		infraOptions := entities.ContainerCLIOpts{ImageVolume: "bind"}
+		pullPolicy := config.PullPolicyNewer
+		// This ensures the image is the image store
+		pullOptions := &libimage.PullOptions{}
+		pullOptions.AuthFilePath = options.Authfile
+		pullOptions.CertDirPath = options.CertDir
+		pullOptions.SignaturePolicyPath = options.SignaturePolicy
+		pullOptions.Writer = writer
+		pullOptions.Username = options.Username
+		pullOptions.Password = options.Password
+		pullOptions.InsecureSkipTLSVerify = options.SkipTLSVerify
+
+		pulledImages, err := ic.Libpod.LibimageRuntime().Pull(ctx, imagePull, pullPolicy, pullOptions)
```

Comment:
This seems a bit redundant with the pull code below. Can we move that into a separate function that will do the config->pullOptions conversion and perform the pull? Probably need some arguments such as the image and the pull policy.

Reply:
good idea, implemented.